### PR TITLE
UX improvements

### LINKED ISSUE #28: Logger references undefined env variable
When building this project with the included build script, the logger component throws reference errors for an undefined env variable.

Transcend didn't notice this issue as we were importing the UI module & building using a different build script that included additional environment variables.

This is fixed in https://github.com/transcend-io/consent-manager-ui/pull/25

### PR DESCRIPTION
**Pull request by eligrey**
Do-Not-Track now entirely suppresses displaying the UI from `transcend.autoShowConsentManager()` (called by `data-prompt="#"`). Global Privacy Control also suppresses the UI for users with privacy regimes (such as CPRA) that cause `SaleOfInfo` to be the only configurable tracking purpose.

## Related Issues

- Blocked by https://github.com/transcend-io/logger/pull/11
- Blocked by https://github.com/transcend-io/website-2.0/pull/41
- Fixes https://github.com/transcend-io/consent-manager-ui/issues/28

## Security Implications

_[none]_

## System Availability

_[none]_

### CODE REVIEW

**Review by anotherminh — DISMISSED** [review]

File: src/components/App.tsx
Comment (anotherminh) [inline src/components/App.tsx:81]: I'm not sure about this actually - if we're going to add this I would also write docs to make sure we explain to the customer why it's not working as they expected

@bencmbrook what do you think?

File: src/init.ts
Comment (eligrey) [inline src/init.ts:30]:
```suggestion
export const privacySignals = new Set<UserPrivacySignal>();
```

**Comment by eligrey** [comment]
Holding off on approval from @bencmbrook before merging this

File: src/components/BottomMenu.tsx
Comment (bencmbrook) [inline src/components/BottomMenu.tsx:88]: Can we split this functionality into a separate PR? This needs a bit of design review, as navigating to `CompleteOptions` doesn't make much sense anymore. Given the opt out is being set here, it should be confirmed in the following view. This view is a slightly confusing place to land, as there's a save button.

**Review by bencmbrook — CHANGES_REQUESTED** [review]

File: src/components/App.tsx
Comment (eligrey) [reply src/components/App.tsx:81]: Those are good points. I definitely expect to need to update this approach as the legal field changes.

I'm going to refactor the auto-prompt suppression logic to be a bit more clear and easier to maintain. My next commit will check `DNT || GPC_BANNER_SUPPRESSION_REGIMES. includes(regime)` where that constant is just `['CPRA']` for now

File: src/components/BottomMenu.tsx
Comment (eligrey) [reply src/components/BottomMenu.tsx:88]: Ok, fine to split this out 👍

Comment (bencmbrook) [reply src/components/BottomMenu.tsx:88]: Thank you! Mind also opening a ticket in Height for the design review?

Comment (eligrey) [reply src/components/BottomMenu.tsx:88]: Done: https://transcend.height.app/T-14275

File: src/components/App.tsx
Comment (eligrey) [reply src/components/App.tsx:90]: I've updated my approach to dynamically resolve the current consent selections, and only suppress auto-prompting with GPC when SaleOfInfo is the only available consent selection

Comment (eligrey) [reply src/components/App.tsx:90]: Once we have https://github.com/transcend-io/website-2.0/pull/41 merged, I will have a URL to point to for the docs regarding this behavior as well.

File: src/components/BottomMenu.tsx
Comment (bencmbrook) [reply src/components/BottomMenu.tsx:86]: Thanks!

File: src/components/BottomMenu.tsx
Comment (bencmbrook) [inline src/components/BottomMenu.tsx:87]:
```suggestion
            onClick={() => handleSetViewState(ViewState.CompleteOptions)}
```

Reverts to make this a non-change. It doesn't actually matter but technically a different return type here

File: src/components/App.tsx
Comment (eligrey) [reply src/components/App.tsx:90]: I added a docs link, dependent on the above PR merging

**Review by bencmbrook — DISMISSED** [review]
Looks great!

**Review by michaelfarrell76 — APPROVED** [review]

**Pull request merged**
